targets/avrtiny.S

```diff
@@ -1,4 +1,3 @@
-; TODO: remove these in LLVM 15
 #define __tmp_reg__ r16
 #define __zero_reg__ r17
```

Comment:
I assumed these could be removed in LLVM 15. I was wrong, they are still needed (`__tmp_reg__` and `__zero_reg__` are only available in inline assembly). Hence why I just removed the comment.